ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/HealthCheckResourceTest.java

```diff
@@ -89,13 +76,11 @@ public void shouldGetCachedResponse() {
   }
 
   @Test
-  public void shouldRecheckHealthIfCachedResponseExpired() {
+  public void shouldRecheckHealthIfCachedResponseExpired() throws Exception {
     // Given:
-    when(currentTimeSupplier.get())
-        .thenReturn(1000L)  // time when first response is cached
-        .thenReturn(1020L)  // time of second checkHealth()
-        .thenReturn(1020L); // time when second response is cached
+    healthCheckResource = new HealthCheckResource(healthCheckAgent, Duration.ofMillis(10));
     healthCheckResource.checkHealth();
+    Thread.sleep(11);
```

Comment:
1 ms extra wait is too tight? could potentially be flaky? is there a better way to check this by waiting until a condition is met? e.g https://github.com/awaitility/awaitility 

In this case, we would keep fetching the old value out of the cache in a loop and exit out when the underlying method to recheck health has been called once?

Reply:
Thanks for the tip! Updated the test.